fedmsg_meta_umb/greenwave.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (C) 2018 Red Hat, Inc.
+#
+# fedmsg_meta_umb is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# fedmsg_meta_umb is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with fedmsg; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+#
+# Authors:  Giulia Naponiello <gnaponie@redhat.com>
+
+from six.moves.urllib.parse import urlencode
+from fedmsg.meta.base import BaseProcessor
+
+
+class GreenwaveProcessor(BaseProcessor):
+    topic_prefix_re = r'/topic/VirtualTopic\.eng'
+
+    __name__ = 'greenwave'
+    __description__ = 'Greenwave Gating Service'
+    __link__ = 'https://greenwave.engineering.redhat.com/api/v1.0/policies'
+    __obj__ = 'Gating Decisions'
+    __docs__ = 'https://mojo.redhat.com/docs/DOC-1171796'
+    __icon__ = ('https://datagrepper-prod-datanommer.int.open.paas.redhat.com/'
+                'umb/_static/img/icons/greenwave.png')
+
+    @staticmethod
+    def satisfied(msg):
+        try:
+            # https://pagure.io/greenwave/pull-request/100
+            return msg['msg']['policies_satisfied']
+        except KeyError:
+            return msg['msg']['policies_satisified']
+
+    def link(self, msg, **config):
+        subject = msg['msg']['subject']
+        if subject:
+            base = "https://resultsdb-api.engineering.redhat.com/api/v2.0/results"
```

Comment:
We can provide the API link to the policies... https://greenwave.stage.engineering.redhat.com/api/v1.0/policies

Reply:
There is no link to a particular decision in greenwave because you have to POST to get it (and the POST is required because otherwise the querystring becomes too long).

Reply:
Ok, I guess the link to resultsdb makes the most sense then. You can disregard this comment.

Reply:
Should it maybe link to the Resultsdb frontend instead of Resultsdb API though?

Reply:
mmm but then you cannot filter for a specific item and type (like is doing in the next lines)... Or am I missing something?
Example: https://resultsdb-api.engineering.redhat.com/api/v2.0/results?item=setup-2.8.71-7.el7_4&type=brew-build

Reply:
@danc86  I did not understand which other link we can point. Do you have an example?

Reply:
The Resultsdb frontend takes the same filter arguments (and just passes them to the backend). The only difference is you get a nicer human readable table with navigation etc. Equivalent example: https://resultsdb.engineering.redhat.com/results?item=setup-2.8.71-7.el7_4&type=brew-build